### PR TITLE
DBZ-1962 Introduce column.whitelist for the Postgres connector

### PR DESCRIPTION
**Pull request by jhuiting**
https://issues.redhat.com/browse/DBZ-1962

Right now the Debezium Postgres connector just supports `column.blacklist`. Added `column.whitelist` as well which might be convenient in some cases.

### CODE REVIEW

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 1]
```diff
@@ -1755,6 +1755,11 @@ Only alphanumeric characters and underscores should be used.
...
 |An optional comma-separated list of regular expressions that match fully-qualified table identifiers for tables to be excluded from monitoring; any table not included in the blacklist will be monitored. Each identifier is of the form _schemaName_._tableName_. May not be used with `table.whitelist`.
 
+|`column.whitelist`
+|
+|An optional comma-separated list of regular expressions that match the fully-qualified names of columns that should be included in change event message values. Fully-qualified names for columns are of the form _schemaName_._tableName_._columnName_
+
+
 |`column.blacklist`
 |
 |An optional comma-separated list of regular expressions that match the fully-qualified names of columns that should be excluded from change event message values. Fully-qualified names for columns are of the form _schemaName_._tableName_._columnName_
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:1765]: Can you add a sentence "May not be used with `column.whitelist`.

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java [hunk 2]
```diff
@@ -768,6 +769,26 @@ public void shouldTakeBlacklistFiltersIntoAccount() throws Exception {
         assertNoRecordsToConsume();
     }
 
+    @Test
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:772]: Could you add the `@FixFor` annotation?

Comment (gunnarmorling) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:772]: Could you also add another test case which ensures that a whitelisted column can be subject to column masking or truncating?

File: debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java [hunk 3]
```diff
@@ -181,6 +181,20 @@ public static DecimalHandlingMode parse(String value, String defaultValue) {
...
+     * and change messages. The exact form of fully qualified names for columns might vary between connector types.
+     * For instance, they could be of the form {@code <databaseName>.<tableName>.<columnName>} or
+     * {@code <schemaName>.<tableName>.<columnName>} or {@code <databaseName>.<schemaName>.<tableName>.<columnName>}.
+     */
+    public static final Field COLUMN_WHITELIST = Field.create("column.whitelist")
+            .withDisplayName("Include Columns")
+            .withType(Type.STRING)
+            .withWidth(Width.LONG)
+            .withImportance(Importance.MEDIUM)
+            .withValidation(RelationalDatabaseConnectorConfig::validateColumnBlacklist)
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java:195]: This validation should go to `COLUMN_BLACKLIST`, right? Would be nice to add a test asserting that validation.

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java [hunk 4]
```diff
@@ -768,6 +769,26 @@ public void shouldTakeBlacklistFiltersIntoAccount() throws Exception {
...
+    public void shouldTakeColumnWhitelistFilterIntoAccount() throws Exception {
+        String setupStmt = SETUP_TABLES_STMT +
+                "CREATE TABLE s1.b (pk SERIAL, aa integer, bb integer, PRIMARY KEY(pk));" +
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:775]: Seems like this table s1.b is not used later on?

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 5]
```diff
@@ -593,7 +593,7 @@ Therefore, we interpret this key as describing the row in the `public.customers`
 
 [NOTE]
 ====
-Although the `column.blacklist` configuration property allows you to remove columns from the event values, all columns in a primary or unique key are always included in the event's key.
+Although the `column.blacklist` and `column.whitelist` configuration properties allow you to remove columns from the event values, all columns in a primary or unique key are always included in the event's key.
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:596]:
```suggestion
Although the `column.blacklist` and `column.whitelist` configuration properties allow you to capture only a subset of table columns, all columns in a primary or unique key are always included in the event's key.
```

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java (hunk 2 above)
Comment (jhuiting) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:772]: Done, extended the test

File: debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java (hunk 3 above)
Comment (jhuiting) [reply debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java]: Done, and added a test

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java (hunk 4 above)
Comment (jhuiting) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java]: Removed

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc (hunk 5 above)
Comment (jhuiting) [reply documentation/modules/ROOT/pages/connectors/postgresql.adoc]: Done

**Comment by jhuiting** [comment]
@gunnarmorling Fixed your comments and added the extra test you requested. Please check again.

**Comment by gunnarmorling** [comment]
Thanks a lot, great work! Added one small commit for adding the option descriptions and applied. Will be part in 1.2.0.Beta1 in three weeks.

**Comment by gunnarmorling** [comment]
@jhuiting btw. is your organization using Debezium in production already? If so, would you be interested in being listed as a user here: https://debezium.io/community/users/?

In that case, all it'd take is a PR for https://github.com/debezium/debezium.github.io/blob/develop/community/users.asciidoc. Thanks a lot for your consideration!

**Comment by jhuiting** [comment]
@gunnarmorling We're not in production yet, but when we are I'll open a PR. Thanks for the suggestion.

**Pull request merged**